src/core/src/bootstrap/EnvLayer.py

```diff
@@ -65,7 +65,7 @@ def get_package_manager(self):
         ret = None
 
         # choose default - almost surely one will match.
-        for b in ('apt-get', 'yum', 'zypper'):
+        for b in ('apt-get', 'yum', 'zypper', 'tdnf'):
```

Comment:
Check for Mariner first separately, and then insider tdnf -> exit out of here.

Let the rest remain the same.